ccl/changefeedccl: TestChangefeedNemeses failed

SHA: https://github.com/cockroachdb/cockroach/commits/e1c1a9ae765730782f3adc2298a76d6a07ad055d

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestChangefeedNemeses PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1626697&tab=buildLog

```
=== RUN   TestChangefeedNemeses
--- FAIL: TestChangefeedNemeses (9.22s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestChangefeedNemeses394664955
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestChangefeedNemeses/cloudstorage
test logs left over in: /tmp/logTestChangefeedNemeses394664955
    --- FAIL: TestChangefeedNemeses/cloudstorage (1.57s)
        nemeses_test.go:38: unexpected file: /tmp/TestChangefeedNemeses_cloudstorage199919238/0/2019-12-05/201912050650021303067710000000000.RESOLVED225278032
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest.(*cloudFeed).walkDir
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest/testfeed.go:645
            path/filepath.walk
            	/usr/local/go/src/path/filepath/path.go:358
            path/filepath.walk
            	/usr/local/go/src/path/filepath/path.go:382
            path/filepath.walk
            	/usr/local/go/src/path/filepath/path.go:382
            path/filepath.Walk
            	/usr/local/go/src/path/filepath/path.go:404
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest.(*cloudFeed).Next
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest/testfeed.go:610
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest.noteFeedMessage
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest/nemeses.go:628
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest.logEvent.func1
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest/nemeses.go:499
            github.com/cockroachdb/cockroach/pkg/util/fsm.Transitions.apply
            	/go/src/github.com/cockroachdb/cockroach/pkg/util/fsm/fsm.go:106
            github.com/cockroachdb/cockroach/pkg/util/fsm.(*Machine).ApplyWithPayload
            	/go/src/github.com/cockroachdb/cockroach/pkg/util/fsm/fsm.go:135
            github.com/cockroachdb/cockroach/pkg/util/fsm.(*Machine).Apply
            	/go/src/github.com/cockroachdb/cockroach/pkg/util/fsm/fsm.go:129
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest.RunNemesis
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest/nemeses.go:215
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedNemeses.func2
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/nemeses_test.go:36
            github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cloudStorageTest.func1
            	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:331
            testing.tRunner
            	/usr/local/go/src/testing/testing.go:865
            runtime.goexit
            	/usr/local/go/src/runtime/asm_amd64.s:1337

```

--- a/pkg/blobs/local_storage.go
+++ b/pkg/blobs/local_storage.go
@@ -73,7 +73,9 @@ func (l *LocalStorage) WriteFile(filename string, content io.Reader) (err error)
 	// - it avoids a cross-filesystem rename in the common case.
 	//   (There can still be cross-filesystem renames in very
 	//   exotic edge cases, hence the use fileutil.Move below.)
-	tmpFile, err := ioutil.TempFile(targetDir, filepath.Base(fullPath))
+	// See the explanatory comment for ioutil.TempFile to understand
+	// what the "*" in the suffix means.
+	tmpFile, err := ioutil.TempFile(targetDir, filepath.Base(fullPath)+"*.tmp")
 	if err != nil {
 		return errors.Wrap(err, "creating temporary file")
 	}

